docs/DeveloperGuide.md

```diff
@@ -220,7 +220,35 @@ The following activity diagram shows the process when a user calls the `EditProf
 
 #### Design Considerations
 
-The design considerations for the edit commands and the add commands are largely similar, please refer to the [Design Considerations](#design-considerations) for the add commands for more details. 
+The design considerations for the edit commands and the add commands are largely similar, please refer to the [Design Considerations](#design-considerations) for the add commands for more details.
+
+### View Command
+
+#### Description
+
+In this section, we will describe how our view commands are implemented. In NUScheduler, there are two variants of view commands, namely the `ViewProfileCommand` and the `ViewEventCommand`. `ViewProfileCommand` is used to view a list of `Profile`s, whereas `ViewEventCommand` is used to view a list of `Event`s.
+
+Since both `ViewProfileCommand` and `ViewEventCommand` are implemented in a similar manner, we will be using the `ViewProfileCommand` to illustrate the implementation of view commands.
+
+The `ViewProfileCommand` extends the `ProfileCommand` abstract class. `ProfileCommand` is an abstract class which extends the `Command` class. `ViewProfileCommand` overrides the `Command#execute` method, to view existing profiles when called.
+
+#### Implementation
+
+1. After the user command is parsed by `NuSchedulerParser`, a `Command` object (more precisely, a `ViewCommand` object) will be returned to the `LogicManager`.
+2. The `LogicManager` will then call `Command#execute` of the `ViewCommand`, passing the `Model` object as parameter.
+3. During the execution of the `ViewCommand`, `Model#updateFilteredProfileList` will be called. The method takes in a `Predicate`, in this case the `Predicate` returns `true` for all inputs. This is to allow all `Profile`s to be listed.
+4. `Model#updateFilteredProfileList` method will then update the `FilteredList` to contain all `Profile`s, which will then be reflected on the terminal.
+5. At the end of method, a `CommandResult` object will be returned which will be used to indicate a successful execution of the command in the display.
+
+The following sequence diagram shows what happens when `ViewCommand` gets executed.
+
+![ViewProfileCommandSequenceDiagram](images/commands/ViewProfileCommandSequenceDiagram.png)
+
+The following activity diagram summarizes what happens when a user executes a view command.
+
+![ViewProfileCommandActivityDiagram](images/commands/ViewProfileCommandActivityDiagram.png)
+
+#### Description
```

Comment:
I think you might have accidentally added another description here.

Reply:
Oh yes good catch!